### PR TITLE
[MSPAINT] Add Zoomed and UnZoomed and use them

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Our code became too complicated. Try to simplify.
JIRA issue: [CORE-17931](https://jira.reactos.org/browse/CORE-17931)

## Proposed changes

- Define `Zoomed` and `UnZoomed` helper functions.
- Use them.

### CODE REVIEW

File: base/applications/mspaint/imgarea.cpp [hunk 1]
```diff
@@ -76,26 +76,26 @@ LRESULT CImgAreaWindow::OnSize(UINT nMsg, WPARAM wParam, LPARAM lParam, BOOL& bH
                0,
                0, 3, 3, TRUE);
     sizeboxCenterTop.MoveWindow(
-               imgXRes * toolsModel.GetZoom() / 2000 + 3 * 3 / 4,
+               toolsModel.Zoomed(imgXRes / 2) + 3 * 3 / 4,
```
Comment (HBelusca) [inline base/applications/mspaint/imgarea.cpp:79]: Would it be more correct to do `toolsModel.Zoomed(imgXRes) / 2` (plus the rest) instead?

File: base/applications/mspaint/winproc.cpp [hunk 2]
```diff
@@ -39,7 +39,7 @@ zoomTo(int newZoom, int mouseX, int mouseY)
     toolsModel.SetZoom(newZoom);
 
     selectionWindow.ShowWindow(SW_HIDE);
-    imageArea.MoveWindow(3, 3, imageModel.GetWidth() * toolsModel.GetZoom() / 1000, imageModel.GetHeight() * toolsModel.GetZoom() / 1000, FALSE);
+    imageArea.MoveWindow(3, 3, toolsModel.Zoomed(imageModel.GetWidth()) / 1000, toolsModel.Zoomed(imageModel.GetHeight()), FALSE);
```
Comment (HBelusca) [inline base/applications/mspaint/winproc.cpp:42]:
```suggestion
    imageArea.MoveWindow(3, 3, toolsModel.Zoomed(imageModel.GetWidth()), toolsModel.Zoomed(imageModel.GetHeight()), FALSE);
```

File: base/applications/mspaint/winproc.cpp [hunk 3]
```diff
@@ -39,7 +39,7 @@ zoomTo(int newZoom, int mouseX, int mouseY)
     toolsModel.SetZoom(newZoom);
 
     selectionWindow.ShowWindow(SW_HIDE);
-    imageArea.MoveWindow(3, 3, imageModel.GetWidth() * toolsModel.GetZoom() / 1000, imageModel.GetHeight() * toolsModel.GetZoom() / 1000, FALSE);
+    imageArea.MoveWindow(3, 3, Zoomed(imageModel.GetWidth()) / 1000, Zoomed(imageModel.GetHeight()), FALSE);
```
Comment (HBelusca) [inline base/applications/mspaint/winproc.cpp:42]:
```suggestion
    imageArea.MoveWindow(3, 3, Zoomed(imageModel.GetWidth()), Zoomed(imageModel.GetHeight()), FALSE);
```

File: base/applications/mspaint/winproc.cpp (hunk 2 above)
Comment (katahiromz) [reply base/applications/mspaint/winproc.cpp]: OK.

File: base/applications/mspaint/winproc.cpp (hunk 3 above)
Comment (katahiromz) [reply base/applications/mspaint/winproc.cpp]: OK.

File: base/applications/mspaint/imgarea.cpp (hunk 1 above)
Comment (katahiromz) [reply base/applications/mspaint/imgarea.cpp]: Fixed.

File: base/applications/mspaint/common.h [hunk 4]
```diff
@@ -11,4 +11,6 @@
 
 /* FUNCTIONS ********************************************************/
 
-extern BOOL zoomTo(int, int, int);
+BOOL zoomTo(int, int, int);
```
Comment (HBelusca) [inline base/applications/mspaint/common.h:14]:
```suggestion
BOOL zoomTo(int newZoom, int mouseX, int mouseY);
```

File: base/applications/mspaint/common.h [hunk 5]
```diff
@@ -11,4 +11,6 @@
...
+int Zoomed(int xy);
+int UnZoomed(int xy);
```
Comment (HBelusca) [inline base/applications/mspaint/common.h:16]: I think that these two functions could be implemented in this file and marked as inlined.

File: base/applications/mspaint/common.h (hunk 4 above)
Comment (katahiromz) [reply base/applications/mspaint/common.h]: OK.

File: base/applications/mspaint/common.h (hunk 5 above)
Comment (katahiromz) [reply base/applications/mspaint/common.h]: Needs a global variable and its class definition.

Comment (HBelusca) [reply base/applications/mspaint/common.h]: If common.h gets included after inclusion of toolsModel in the source files that use it, then it should work (without the need for common.h itself to include that other .h) 🤔

Comment (katahiromz) [reply base/applications/mspaint/common.h]: Fixed.

File: base/applications/mspaint/precomp.h [hunk 6]
```diff
@@ -43,5 +42,6 @@
 #include "toolsettings.h"
 #include "toolsmodel.h"
 #include "winproc.h"
+#include "common.h"
```
Comment (JoachimHenze) [inline base/applications/mspaint/precomp.h:45]: What was the exact reason you had to move this include?

Comment (katahiromz) [reply base/applications/mspaint/precomp.h:45]: Nothing.

Comment (katahiromz) [reply base/applications/mspaint/precomp.h:45]: To use the global variable, its type should be predefined.

Comment (katahiromz) [reply base/applications/mspaint/precomp.h:45]: https://github.com/reactos/reactos/pull/4188#discussion_r775185392

**Review by roybaer — APPROVED** [review]
Yes, this looks like sensible refactoring.
I would probably have used macros out of habit, but `static inline` functions should work at least equally well.

**Pull request merged**